src/main/java/com/example/demo/controller/planner/TripPlannerController.java

```diff
@@ -0,0 +1,239 @@
+package com.example.demo.controller.planner;
+
+import com.example.demo.dto.planner.trip.TripRequestDTO;
+import com.example.demo.dto.planner.trip.TripResponseDTO;
+import com.example.demo.dto.users.user.UserIdsDTO;
+import com.example.demo.entity.planner.Trip;
+import com.example.demo.exception.InvalidUserException;
+import com.example.demo.service.jwt.JwtCheckService;
+import com.example.demo.service.planner.TripDateNotFoundException;
+import com.example.demo.service.planner.TripNotFoundException;
+import com.example.demo.service.planner.TripPlannerService;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * TripPlannerController | 이 컨트롤러는 여행 계획과 관련된 요청을 처리합니다.
+ * 사용자가 여행 계획을 생성, 조회, 수정, 삭제, 공유할 수 있도록 합니다.
+ * @see TripPlannerService 여행 계획과 관련된 비즈니스 로직을 처리하는 서비스 클래스
+ * @see TripRequestDTO 여행 계획 생성 요청 데이터를 담고 있는 DTO
+ * @see TripResponseDTO 여행 계획 응답 데이터를 담고 있는 DTO
+ * @see JwtCheckService JWT 토큰을 검증하여 요청의 유효성을 검사하는 서비스
+ * @see TripNotFoundException 여행 계획이 존재하지 않을 때 발생하는 예외
+ * @author minjeong
+ */
+@RestController()
+@RequestMapping("/planner")
+@RequiredArgsConstructor
+@Slf4j
+public class TripPlannerController {
+
+    private final JwtCheckService jwtCheckService;
+    private final TripPlannerService tripPlannerService;
+
+    /**
+     * 모든 여행 계획을 조회합니다.
+     * 이 메서드는 JWT 토큰을 확인한 후 모든 여행 계획을 가져옵니다.
+     * 만약 여행 계획이 없다면, `NO_CONTENT` 상태를 반환합니다.
+     *
+     * @param request  HTTP 요청 객체
+     * @param response HTTP 응답 객체
+     * @return 여행 계획 리스트와 상태 코드를 포함한 응답 엔티티
+     */
+    @GetMapping("/trip")
+    public ResponseEntity<List<TripResponseDTO>> getAllTrips(HttpServletRequest request, HttpServletResponse response) {
+        jwtCheckService.checkJwt(request, response);
+        try {
+            List<Trip> trips = tripPlannerService.getAllTrips();
+            List<TripResponseDTO> responseDTOs = trips.stream()
+                    .map(TripResponseDTO::fromEntity)
+                    .collect(Collectors.toList());
```

Comment:
entity list -> dto list 변환할 때, Collections 클래스나 lamda함수를 사용해서 Controller 클래스 내에서 작업하는것도 좋지만, 함수화를 할 수도 있을거 같은데 어떻게 생각하세요?

아래는 제가 다른 프로젝트에서 사용한 코드입니다.

```suggestion
             
package com.boosting.global.converter;

import java.util.ArrayList;
import java.util.List;
import java.util.function.Function;

public class DtoConverter {

    /**
     * 엔티티 리스트를 DTO 리스트로 변환하는 메서드
     *
     * @param entityList 변환할 엔티티 리스트
     * @param converter  각 엔티티를 DTO로 변환하는 함수
     * @param <T>        엔티티 타입
     * @param <R>        DTO 타입
     * @return 변환된 DTO 리스트
     */
    public static <T, R> List<R> convertEntityListToDtoList(List<T> entityList, Function<T, R> converter) {
        List<R> dtoList = new ArrayList<>();
        entityList.forEach(item -> dtoList.add(converter.apply(item)));
        return dtoList;
    }
}
         
```

Reply:
좋습니다!